test/e2e/tests/settings/4byte-directory.spec.js

```diff
@@ -27,27 +25,23 @@ describe('4byte setting', function () {
         const contractAddress = await contractRegistry.getContractAddress(
           smartContract,
         );
-        await unlockWallet(driver);
+        await logInWithBalanceValidation(driver);
 
         // deploy contract
         await openDapp(driver, contractAddress);
 
         // wait for deployed contract, calls and confirms a contract method where ETH is sent
-        await driver.delay(largeDelayMs);
         await driver.clickElement('#depositButton');
 
-        await driver.waitForSelector({
-          css: 'span',
-          text: 'Deposit initiated',
-        });
-
-        await driver.waitUntilXWindowHandles(3);
         await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-        const actionElement = await driver.waitForSelector({
-          css: '.confirm-page-container-summary__action__name',
+        await driver.waitForSelector({
+          tag: 'span',
           text: 'Deposit',
         });
-        assert.equal(await actionElement.getText(), 'DEPOSIT');
```

Comment:
asserting an element by it's inner value to match an expect value, opens the door to race conditions. We should find the element by its value directly. I've also added the 2 conditions we want (deposit is there, and contract interaction is not there)